src/WpfMath/Boxes/CharBox.cs

```diff
@@ -35,7 +35,7 @@ internal GlyphRun GetGlyphRun(double scale, double x, double y)
 
         public override void RenderTo(IElementRenderer renderer, double x, double y)
         {
-            var color = this.Foreground ?? Brushes.Black;
+            var color = this.Foreground ??(renderer.Foreground?? Brushes.Black);
```

Comment:
Instead of adding the complexity here (by adding a public `Foreground` property), I think we could reduce it: let's remove the `Brushes.Black` altogether, and let the renderer to define their default colors as they wish.